lib/ex_activecampaign/api_v1.ex

```diff
@@ -0,0 +1,236 @@
+defmodule ExActivecampaign.ApiV1 do
+  @moduledoc """
+    A wrapper for ActiveCampaign API V1 operations.
+  """
+
+  @doc """
+  Send a GET request to the ActiveCampaign API
+  """
+  def get(url, query_params \\ %{}, headers \\ []) do
+    call(url, :get, nil, query_params, headers)
+  end
+
+  @doc """
+  Send a POST request to the ActiveCampaign API
+  """
+  def post(url, body \\ nil, query_params \\ %{}, headers \\ []) do
+    call(url, :post, body, query_params, headers)
+  end
+
+  defp call(url, method, body, query_params, headers) do
+    HTTPoison.request(
+      method,
+      url |> clean_url,
+      body |> encode(content_type(headers)),
+      headers |> clean_headers,
+      query_params |> clean_params
+    )
+    |> case do
+      {:ok, %{body: raw_body, status_code: code, headers: headers}} ->
+        {code, raw_body, headers}
+
+      {:error, %{reason: reason}} ->
+        {:error, reason, []}
+    end
+    |> decode
+  end
+
+  @doc """
+  Extract the content type of the headers
+
+  ## Examples
+
+      iex> ExActivecampaign.ApiV1.content_type({:ok, %{"email" => "johndoe@example.com"}, [{"Server", "GitHub.com"}, {"Content-Type", "application/xml; charset=utf-8"}]})
+      {:ok, %{"email" => "johndoe@example.com"}, "application/x-www-form-urlencoded"}
+
+      iex> ExActivecampaign.ApiV1.content_type([])
+      "application/x-www-form-urlencoded"
+
+      iex> ExActivecampaign.ApiV1.content_type([{"Content-Type", "plain/text"}])
+      "application/x-www-form-urlencoded"
+
+      iex> ExActivecampaign.ApiV1.content_type([{"Content-Type", "application/xml; charset=utf-8"}])
+      "application/x-www-form-urlencoded"
+
+      iex> ExActivecampaign.ApiV1.content_type([{"Server", "GitHub.com"}, {"Content-Type", "application/xml; charset=utf-8"}])
+      "application/x-www-form-urlencoded"
+  """
+  def content_type({status, body, headers}), do: {status, body, content_type(headers)}
+  def content_type([]), do: "application/x-www-form-urlencoded"
+  def content_type([{"Content-Type", _val} | _]), do: "application/x-www-form-urlencoded"
+  def content_type([_ | tail]), do: tail |> content_type
```

Comment:
I'm a little unsure what this is doing. Is there any case where it won't return "application/x-www-form-urlencoded"?

Reply:
Yeah, I could clean this up a fair bit tbf. I guess I got hooked into a bit of premature optimisation, accounting for features we aren't even using.